src/results/aggregator.rs

```diff
@@ -143,11 +151,46 @@ pub async fn aggregate(
         }
     }
 
-    let results = result_map.into_values().collect();
+    let mut blacklist_map: HashMap<String, SearchResult> = HashMap::new();
+    filter_with_lists(
+        &mut result_map,
+        &mut blacklist_map,
+        &file_path(FileType::BlockList)?,
+    )?;
+
+    filter_with_lists(
+        &mut blacklist_map,
+        &mut result_map,
+        &file_path(FileType::AllowList)?,
+    )?;
+
+    drop(blacklist_map);
+
+    let results: Vec<SearchResult> = result_map.into_values().collect();
 
     Ok(SearchResults::new(
         results,
         query.to_string(),
         engine_errors_info,
     ))
 }
+
+fn filter_with_lists(
```

Comment:
I think we could add all the tests at the end once we are done with everything by putting a seperate PR for this. I think this would be better. What do you say ? :slightly_smiling_face: .

Reply:
> I think we could add all the tests at the end once we are done with everything by putting a separate PR for this. I think this would be better. What do you say ? 🙂 .

That's fine. Let's go ahead and do it that way for now

> Also, to be honest, I am not really the right person for writing tests because I am not good at it 😂 . So I think maybe we might need someone who has a good amount of experience to write tests. What do you say ?

I can help write some unit tests